### ISSUE TITLE
Multi-gpu spread not only memory, but perfomance also

### ISSUE THREAD

**Issue by NameRX** [jcjohnson/neural-style#369]
Hello! Is it possible to spread along several GPUs not only memory to produce a hi-res image, but also GPU's performance?

If it is not, maybe it is possible to use all GPUs to make single lo-res image simultaneously for faster rendering?

**Comment by thedonquixotic** [comment]
I am also curious about this. I've been running Deep Style on my home gpu, and I'd like to try hooking up a bunch of spare gpus I have lying around from an old mining rig, into a computer cluster, to run Deep Style at higher resolutions and with more memory. 

Is this possible?

**Comment by shinobiultra** [comment]
Yes! You can use `-gpu_strategy` to decide, which layers to run on which gpu.

From official neural-style git:

> For example in a server with four GPUs, you can give the flag -gpu 0,1,2,3 to process on GPUs 0, 1, 2, and 3 in that order; by also giving the flag -multigpu_strategy 3,6,12 you indicate that the first two layers should be computed on GPU 0, layers 3 to 5 should be computed on GPU 1, layers 6 to 11 should be computed on GPU 2, and the remaining layers should be computed on GPU 3. You will need to tune the -multigpu_strategy for your setup in order to achieve maximal resolution.
